flow/e2e/s3/s3_helper.go

```diff
@@ -24,28 +24,48 @@ type S3TestHelper struct {
 	prefix     string
 }
 
-func NewS3TestHelper(switchToGCS bool) (*S3TestHelper, error) {
-	credsPath := os.Getenv("TEST_S3_CREDS")
-	bucketName := "peerdb-test-bucket"
-	if switchToGCS {
+type S3Environment int
+
+const (
+	Aws S3Environment = iota
+	Gcs
+	Minio
+)
+
+func NewS3TestHelper(s3environment S3Environment) (*S3TestHelper, error) {
+	var config utils.S3PeerCredentials
+	var endpoint string
+	var credsPath string
+	var bucketName string
+	switch s3environment {
+	case Aws:
+		credsPath = os.Getenv("TEST_S3_CREDS")
+		bucketName = "peerdb-test-bucket"
```

Comment:
not sure why this has to be in code?
if it does, make it a map?

Reply:
Hard to be a map when it has specific behavior (loading files for s3/gcs, using env for minio). This change is fleshing out two-state bool into tri-state enum since we already had this logic for aws/gcs, but now we support minio